### PR TITLE
[backend] feature(telemetry): use share and download events from ES for service counters #1028

### PR DESCRIPTION
**Pull request by carinelebas**
# Context
> Briefly describe the purpose and context of this PR. Include relevant screenshots or screen recordings for the product team.
We want to use counter for share and download from elastic search instead of postgresql. We also need to populate elastic search with past events with the right number of events per document. 
Clean up/removing the DB columns will be done in another PR.

For custom dashboards, csv feeds, openaev scenarii : 
- test downloads are well increased after downloading the files (refresh page to be sure a new fetch from db is done) 
- when logged, test shares are well increased after downloading the files (refresh page to be sure a new fetch from db is done) 
- on public paged (unlogged), test shares are well increased after downloading the files (refresh page to be sure a new fetch from db is done)
- When starting the server, check the right number of download and share events is created in elastic search

## Related issues

- Related to #1028

# Checklist

## Quality

- [x] I consider this work complete and ready for review
- [x] I tested all features and edge cases
- [x] I refactored code where necessary to improve quality
- [x] I made sure my code meets development guidelines
- [x] I performed a self-review of my code

## Testing

- [x] I wrote test cases for the relevant use cases (unit, integration or e2e tests)
- [x] (Bug fixes only) I added tests that reproduce and verify the fix

### What tests have been made

- [x] Unit tests
- [ ] Integration tests
- [ ] E2E tests
- [x] Local manual tests

# Additional information
> Any additional context, dependencies, or concerns reviewers should know about.

### CODE REVIEW

File: apps/portal-api/src/es-migrations/1759740338898-create_share_download_events.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+  const already_created_events = new Map(
+    buckets.map((bucket) => [bucket.key, bucket.doc_count])
+  );
+
+  return downloaded_docs.flatMap((doc) => {
+    const alreadyCreated = already_created_events.get(doc.id) || 0;
+    const totalCount = event_type ? doc.download_number : doc.share_number;
+    const newEventsCount = Math.max(0, totalCount - alreadyCreated);
+
+    const service =
```
[the remark this answers is not in the document]
Comment (carinelebas) [reply apps/portal-api/src/es-migrations/1759740338898-create_share_download_events.ts:56]: it's only a script, I don't think it requires a lot of maintenance ? Also, we should keep dependencies to a minimum in es scripts to avoid dragging the whole codebase when it's compiled.

File: apps/portal-api/src/es-migrations/1759740338898-create_share_download_events.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+      doc.type === 'csv_feed'
+        ? TelemetryEventService.INTEGRATION_FEEDS_LIBRARY
+        : doc.type === 'custom_dashboard'
+          ? TelemetryEventService.CUSTOM_DASHBOARDS_LIBRARY
+          : TelemetryEventService.OPENAEV_SCENARIOS_LIBRARY;
+
+    const serviceType =
+      doc.type === 'csv_feed' ? TelemetryEventServiceType.CSV_FEEDS : undefined;
```
[the remark this answers is not in the document]
Comment (carinelebas) [reply apps/portal-api/src/es-migrations/1759740338898-create_share_download_events.ts:64]: For telemetry, csv_feeds is one type of "integration feed"(later we can add taxii feeds etc..). For other services (custom dashboards, openaev scenarii), there is no need for a service type, the service itself is enough.

File: apps/portal-api/src/modules/services/custom-dashboards/custom-dashboards.resolver.ts [hunk 3]
```diff
@@ -79,10 +73,9 @@ const resolvers: Resolvers = {
       );
     },
     customDashboard: async (_, { id }, context) =>
-      loadDocumentById(
+      CustomDashboardsApp.loadCustomDashboard(
```
[the remark this answers is not in the document]
Comment (hervyt) [reply apps/portal-api/src/modules/services/custom-dashboards/custom-dashboards.resolver.ts:76]: Yes, but the csvFeedsApp starts with lower letter... Maybe we can add a tech pain to harmonize everything ?

File: apps/portal-api/src/modules/services/document/document-download-endpoint.ts [hunk 4]
```diff
@@ -121,7 +121,7 @@ export const documentDownloadEndpoint = (app) => {
           if (attach) {
             res.attachment(document.file_name);
           }
-          await incrementDocumentsDownloads(context, document, trx);
+
```
Comment (hervyt) [inline apps/portal-api/src/modules/services/document/document-download-endpoint.ts:124]: 💯
